hist/hist/src/TF1.cxx

```diff
@@ -2480,8 +2480,12 @@ Double_t TF1::GradientPar(Int_t ipar, const Double_t *x, Double_t eps)
 
 void TF1::GradientPar(const Double_t *x, Double_t *grad, Double_t eps)
 {
-   if (fFormula && fFormula->HasGeneratedGradient())
-      fFormula->GradientPar(x,grad);
+   if (fFormula && fFormula->HasGeneratedGradient()) {
+      // need to allocate a vector here
+      std::vector<double> gvec(fNpar);
+      fFormula->GradientPar(x,gvec);
+      std::copy(gvec.begin(),gvec.end(), grad);
+   }
```

Comment:
The thing with clad is that it expects that the output buffer is zeroed out.

Can you try if this also fixes the problem?
```C++
std::fill(grad, grad + fNpar, 0.0);
```

Then you avoid a memory reallocation here.

Reply:
Good point! I did not know about this requirement, is documented somewhere ? If not, I will add in TFormula